src/mito2/src/read/scan_util.rs

```diff
@@ -568,6 +589,23 @@ pub(crate) fn scan_file_ranges(
         // Reports metrics.
         reader_metrics.observe_rows(read_type);
         reader_metrics.filter_metrics.observe();
-        part_metrics.merge_reader_metrics(&reader_metrics);
+        part_metrics.merge_reader_metrics(reader_metrics);
     }
 }
+
+/// Build the stream of scanning the extension range denoted by the [`RowGroupIndex`].
+#[cfg(feature = "enterprise")]
+pub(crate) async fn scan_extension_range(
+    context: Arc<StreamContext>,
+    index: RowGroupIndex,
+    metrics: PartitionMetrics,
+) -> Result<crate::read::BoxedBatchStream> {
+    use snafu::ResultExt;
+
+    let range = context.input.extension_range(index.index);
+    let reader = range.reader(context.as_ref());
+    reader
+        .read(context, metrics, index)
+        .await
+        .context(crate::error::ScanExternalRangeSnafu)
+}
```

Comment:
We can provide another implementation for `scan_extension_range()` under `not(feature)`. This can simplify the usage of `scan_extension_range()`.
```rust
#[cfg(not(feature = "enterprise"))]
pub(crate) async fn scan_extension_range(
    _context: Arc<StreamContext>,
    _index: RowGroupIndex,
    _metrics: PartitionMetrics,
) -> Result<crate::read::BoxedBatchStream> {
    // return unexpected error
}
```